frontend/src/components/SimpleSelect.tsx

```diff
@@ -27,6 +27,7 @@ export type SimpleSelectOption = {
   isDisabled?: boolean;
   isFavorited?: boolean;
   dataTestId?: string;
+  optionKey?: string; // Used to differentiate the only option with the same key to trigger the one-option hook in the component
```

Comment:
This new key shouldn't be necessary. When there is a single option and there is no selected value then call `onChange` to simulate selecting the option, otherwise do nothing. Currently the selected value is not taken into consideration at all. If a caller requires `onChange` to be called even when supplying the same selected value then there's something wrong with the caller.